shared/core/types/actor.go

```diff
@@ -1,5 +1,15 @@
 package types
 
+// REFACTOR: Evaluate a way to define an interface for the actors which would enable to easily
+//           remove all the `switch actorType` statements in the codebase.
+
+var ActorTypes = []ActorType{
+	ActorType_ACTOR_TYPE_APP,
+	ActorType_ACTOR_TYPE_FISH,
+	ActorType_ACTOR_TYPE_SERVICENODE,
+	ActorType_ACTOR_TYPE_VAL,
+}
+
```

Comment:
I am not sure this is needed. We already have the generated:

```go
// Enum value maps for ActorType.
var (
	ActorType_name = map[int32]string{
		0: "ACTOR_TYPE_UNSPECIFIED",
		1: "ACTOR_TYPE_APP",
		2: "ACTOR_TYPE_SERVICENODE",
		3: "ACTOR_TYPE_FISH",
		4: "ACTOR_TYPE_VAL",
	}
	ActorType_value = map[string]int32{
		"ACTOR_TYPE_UNSPECIFIED": 0,
		"ACTOR_TYPE_APP":         1,
		"ACTOR_TYPE_SERVICENODE": 2,
		"ACTOR_TYPE_FISH":        3,
		"ACTOR_TYPE_VAL":         4,
	}
)
```

in `actor.pb.go` and I think we should rely on them instead. This way it's future proof as well since changes in the enum woudn't require changes here if that makes sense.

Reply:
TL;DR: Yep, nuke the 0.

I'd wrap in a func what you said: 
```go
	for actorTypeNum := range coreTypes.ActorType_name {
		if actorTypeNum == 0 {
			// skipping the zero value (unspecified actor type) since it's not relevant here
			continue
		}
		actorType := coreTypes.ActorType(actorTypeNum)

```
to return a []coreTypes.ActorType

Reply:
Went with what I believ eis the best solution here: [`6733ec4` (#503)](https://github.com/pokt-network/pocket/pull/503/commits/6733ec493c6afaf17fa094afd7599277ca0bb16c)

No redundant type, just use proto directly w/ an if everywhere - very golangy